ynr/apps/uk_results/views/base_views.py

```diff
@@ -26,7 +26,9 @@ class BallotPaperResultsUpdateView(LoginRequiredMixin, FormView):
     def get_form_kwargs(self):
         kwargs = super().get_form_kwargs()
         self.pee = PostExtraElection.objects.get(
-            ballot_paper_id=self.kwargs["ballot_paper_id"]
+            ballot_paper_id=self.kwargs["ballot_paper_id"],
+            # if we try to add results to a cancelled election, throw a 404
```

Comment:
I reckon adding a conditional anywhere this view is linked to, removing the link (or adding a message as to why it's removed) would be helpful.

Reply:
Yep. I've done that here: https://github.com/DemocracyClub/yournextrepresentative/pull/683/files#diff-05649da5568a0db8fb1f8287fe63917dR67 (so it won't get listed in the list of elections with no results) but now you mention it there are probably some other places I haven't thought of where there is a CTA to add results.

Reply:
e.g here: https://github.com/DemocracyClub/yournextrepresentative/blob/master/ynr/apps/candidates/templates/candidates/constituencies.html#L69

A search for `ballot_paper_results_form` turns up a few more.